pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -1384,7 +1384,7 @@ public List<String> getSchemaNames() {
   public void addTable(TableConfig tableConfig)
       throws IOException {
     String tableNameWithType = tableConfig.getTableName();
-    if (getTableConfig(tableNameWithType) != null) {
+    if (hasTable(tableNameWithType)) {
       throw new TableAlreadyExistsException("Table " + tableNameWithType + " already exists");
```

Comment:
nit: I think the error message can be more detailed - 
```
Table already exists in helix ??
```
Or maybe add a debug line in there.

Reply:
Added a more detailed message saying that the table config existed and a suggestion to cleanup the table.